src/Atmos/Model/bc_moisture.jl

```diff
@@ -60,3 +60,56 @@ function atmos_moisture_normal_boundary_flux_second_order!(
     # assumes EquilMoist
     fluxᵀn.moisture.ρq_tot += nρd_q_tot
 end
+
+"""
+    BulkFormulaMoisture(fn) :: MoistureBC
+
+Calculate the net inward moisture flux across the boundary using
+the bulk formula.
+The drag coefficient is `C_q = fn_C_q(state, aux, t, normu_int_tan)`.
+The surface q_tot is `q_tot_sfc = fn_q_tot_sfc(state, aux, t)`.
+Return the flux (in kg m^-2 s^-1).
+"""
+struct BulkFormulaMoisture{FNX, FNM} <: MoistureBC
+    fn_C_q::FNX
+    fn_q_tot_sfc::FNM
+end
+function atmos_moisture_boundary_state!(
+    nf,
+    bc_moisture::BulkFormulaMoisture,
+    atmos,
+    args...,
+) end
+function atmos_moisture_normal_boundary_flux_second_order!(
+    nf,
+    bc_moisture::BulkFormulaMoisture,
+    atmos,
+    fluxᵀn,
+    n⁻,
+    state_sfc⁻,
+    diff_sfc⁻,
+    hyperdiff_sfc⁻,
+    aux_sfc⁻,
+    state_sfc⁺,
+    diff_sfc⁺,
+    hyperdiff_sfc⁺,
+    aux_sfc⁺,
+    bctype,
+    t,
+    state_int⁻,
+    diff_int⁻,
+    aux_int⁻,
+)
+
+    u_int⁻ = state_int⁻.ρu / state_int⁻.ρ
+    u_int⁻_tan = projection_tangential(atmos, aux_int⁻, u_int⁻)
+    normu_int⁻_tan = norm(u_int⁻_tan)
+    C_q = bc_moisture.fn_C_q(state_sfc⁻, aux_sfc⁻, t, normu_int⁻_tan)
+    q_tot_sfc = bc_moisture.fn_q_tot_sfc(state_sfc⁻, aux_sfc⁻, t)
+    q_tot_int = state_int⁻.moisture.ρq_tot / state_int⁻.ρ
+    # NOTE: difference from design docs since normal points outwards
+    fluxᵀn.moisture.ρq_tot -=
+        C_q * state_sfc⁻.ρ * normu_int⁻_tan * (q_tot_sfc - q_tot_int)
+    fluxᵀn.ρ -= C_q * state_sfc⁻.ρ * normu_int⁻_tan * (q_tot_sfc - q_tot_int)
+    fluxᵀn.ρu -= C_q * normu_int⁻_tan * (q_tot_sfc - q_tot_int) .* state_sfc⁻.ρu
```

Comment:
Should we use the density at the surface or at the interior point here? Does it matter? @tapios @akshaysridhar

Reply:
Good question. What would make most sense is using an average density between surface and interior point (for a "central difference approximation"). The problem is that the surface density is probably not accurate because we are not resolving the surface layer. I suggest you write a function for the density, which for now can return either just the interior density, or the mean of interior and surface densities. It does not matter much what you use. If we want something more precise one day, we can then modify that function.

Reply:
@charleskawczynski where would be a good place to add a function like this? boundaryconditions.jl? or relations.jl?

Reply:
I think boundaryconditions.jl is the right place.